src/kdalgorithms.h

```diff
@@ -256,6 +256,23 @@ requires std::is_invocable_r_v<ReturnType, BinaryOperation, ReturnType, ValueTyp
                            std::forward<BinaryOperation>(accumulateFunction));
 }
 
+// -------------------- accumulate_if --------------------
+template <typename Container, typename BinaryOperation, typename UnaryPredicate,
+          typename ReturnType = remove_cvref_t<traits::return_type_of_t<BinaryOperation>>>
+ReturnType accumulate_if(const Container &container, BinaryOperation &&accumulateFunction,
+                         UnaryPredicate &&predicate, ReturnType initialValue = {})
+{
+    auto range = read_iterator_wrapper(container);
+    auto predicateFunction = detail::to_function_object(std::forward<UnaryPredicate>(predicate));
+    auto fn = [&](const ReturnType &subResult, const ValueType<Container> &item) {
+        if (predicateFunction(item))
+            return accumulateFunction(subResult, item);
```

Comment:
If I were to nitpick, this is missing to_function_object - accumulate_if(collection, &QString::append, ...)

(if `append` wasn't overloaded)

This would be nice for member functions that follow the builder pattern
class Type {
    Type with_something(CollectionElement x);
}